File: totosuki/ABC/B/060.py
import time
st = time.time()
A, B, C = map(int, input().split())
x = A
n = 1
while True:
  nt = time.time()
  if nt - st <= 1.92:
    if x % B == C:
      print("YES")
      break
  else:
    print("NO")
    break
  
  n += 1
  x = A * n

File: OCH_Tao/ABC/A/060.js
const S = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"))[0].split(" ");
if ((S[0])[S[0].length - 1] === (S[1])[0] && (S[1])[S[1].length - 1] === (S[2])[0]) {
  console.log("YES");
} else {
  console.log("NO");
}